docs/source/miscellaneous/stubs.rst

```diff
@@ -0,0 +1,610 @@
+#######
+ Stubs
+#######
+
+Stub files are pieces of plain PHP code which only contain declarations without actually runnable
+code. A very basic stub looks like this:
+
+.. code:: php
+
+   /** @var string */
+   const FOO = "foo";
+   /** @var int */
+   const BAR = "bar";
```

Comment:
Can't we use constant types? Or is this only a class thing?

Reply:
No, unfortunately, we can't. :( Coincidentally, I've just mentioned this at the end of this comment: https://github.com/php/php-src/pull/13669#discussion_r1524358975 I'd like if constants had types, but it doesn't seem as straightforward as it sounded before I tried it ^^ and also I didn't really find any use-case besides stubs. :(

Reply:
ACK